libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewEventDispatcher.java

```diff
@@ -167,14 +173,22 @@ void onInstructionListVisibilityChanged(boolean shown) {
 
   SpeechAnnouncement onAnnouncement(SpeechAnnouncement announcement) {
     if (speechAnnouncementListener != null) {
-      return speechAnnouncementListener.willVoice(announcement);
+      String textAnnouncement = speechAnnouncementListener.willVoice(announcement);
+      SpeechAnnouncement announcementToBeVoiced = SpeechAnnouncement.builder().announcement(textAnnouncement).build();
+      return announcementToBeVoiced;
     }
     return announcement;
   }
 
   BannerInstructions onBannerDisplay(BannerInstructions instructions) {
     if (bannerInstructionsListener != null) {
-      return bannerInstructionsListener.willDisplay(instructions);
+      String textBanner = bannerInstructionsListener.willDisplay(instructions);
+      List<BannerComponents> bannerComponents = new ArrayList<>(ONE);
+      bannerComponents.add(instructions.primary().components().get(FIRST).toBuilder().text(textBanner)
```

Comment:
The problem with this is it's going to get rid of any formatting (abbreviations, highway signs). So, if the developer is only overriding some of the instructions, it's still going to remove the formatting from all of them. Also, do we think some advanced developers might take advantage of the abbreviations and/or road signs?

Reply:
> The problem with this is it's going to get rid of any formatting (abbreviations, highway signs). So, if the developer is only overriding some of the instructions, it's still going to remove the formatting from all of them.

Yeah, that's true and the tradeoff that we need to discuss / agree on.

> Also, do we think some advanced developers might take advantage of the abbreviations and/or road signs?

Nope as far as I know.

That being said, I'd prefer to keep it simple for now (until a specific request comes in and at that point we can revisit).